compiler/code-gen/vertex-compiler.cpp

```diff
@@ -1991,25 +1991,36 @@ void compile_callback_of_builtin(VertexAdaptor<op_callback_of_builtin> root, Cod
    * Will be transformed to:
    *   array_map([captured1 = $extern] (auto &&... args) {
    *       return lambda$xxx(captured1, std::forward<decltype(args)>(args)...);
-   *   }), const_array);
+   *   }, const_array);
+   * or in K2 mode:
+   *   co_await array_map([captured1 = $extern] (auto &&... args) [[-> task_t<lambda's return type>]]? {
+   *       [[return lambda$xxx(captured1, std::forward<decltype(args)>(args)...);
+   *       | co_return(co_await lambda$xxx(captured1, std::forward<decltype(args)>(args)...))]]
```

Comment:
Why `[[, ]]`?
As I see output, there are single `[, ]`

Reply:
I tried to use it as a syntax for block of code that support optionality via `?` and disjunction via `|`.

Reply:
It's worse to read, I've spent about 5 mins to parse this syntax :smiley: 
Let's use just English `or` instead of disjunction

Reply:
It's hard to express this in English and keep comment small. So I used Antlr-like syntax :)

Reply:
What `(-> task_t<lambda's return type>)?` stands for?

Reply:
It means that it may have an explicit return type optionally